UmbrellaPipeline/tests/test_grid.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+import numpy as np
+import openmm.app as app
+import openmm.unit as unit
+from UmbrellaPipeline.pathGeneration import Grid, GridNode
+
+pdb = "UmbrellaPipeline/data/step5_input.pdb"
+psf = "UmbrellaPipeline/data/step5_input.psf"
+
+
+def testGridAndNodeFromFiles():
+    grid = Grid.gridFromFiles(
+        pdb=pdb, psf=psf, gridsize=1 * unit.angstrom, vdwradius=1.2 * unit.angstrom
+    )
+    node = grid.nodeFromFiles(pdb=pdb, psf=psf, name="UNL")
+    del grid
+    del node
+    pdbo = app.PDBFile(pdb)
+    psfo = app.CharmmPsfFile(psf)
+    grid = Grid.gridFromFiles(
+        pdb=pdbo, psf=psfo, gridsize=1 * unit.angstrom, vdwradius=1.2 * unit.angstrom
+    )
+    node = grid.nodeFromFiles(pdb=pdb, psf=psf, name="UNL")
+
+
+def testInitializeGridFromNumpy():
+    grid = Grid(
+        grid=np.zeros(shape=(100, 32, 24), dtype=bool),
+        boxlengths=[0 * unit.angstrom, 1 * unit.angstrom, 2 * unit.angstrom],
+    )
+    assert grid.dtype == bool
+    assert grid.x == 100
+    assert grid.b == 1 * unit.angstrom
+
+
+def testInitializeFromValues():
+    grid = Grid(x=23, y=23, z=23, dtype=bool)
+    try:
+        failed = Grid(x=23, y=-23, z=23)
+    except ValueError:
+        pass
```

Comment:
are you expecting that this fails here with a ValueError?

Reply:
jup, since I'm raising a value error in Grid, since the value is not allowed to be negative.